Add order attribution install banner in core analytics overview

ianlin opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Part of #51567, this PR adds the order attribution install banner in the core analytics overview page. The banner will be shown if:
- The `woocommerce-analytics` plugin is not installed yet, and
- The merchant hasn't clicked the `Dismiss` button.

https://github.com/user-attachments/assets/ace17a90-007d-45cc-811d-b712ebd3a300

Note that when clicking `Try it now` on the banner, it will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found as the plugin hasn't been published.

#### Event tracking

This PR also adds three events for tracking:
- `order_attribution_install_banner_viewed`
- `order_attribution_install_banner_clicked`
- `order_attribution_install_banner_dismissed`

#### Next step

- Implement displaying the banner for % of merchants
- Display the banner in other places within WooCommerce Admin
- The changelog entry will be added in the final PR once all the banners are added

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Do not install `woocommerce-analytics` plugin
2. Go to `Analytics -> Overview`
3. It should show the banner
4. Click `Try it now`
5. It will be redirected to https://woocommerce.com/products/woocommerce-analytics, but the page is currently not found
6. Click `Dismiss`
7. The banner is gone
8. Check the option `woocommerce_order_attribution_install_banner_dismissed` in `wp_options` table
9. The value should be `yes`
10. Delete the option `woocommerce_order_attribution_install_banner_dismissed`
11. Go to `Analytics -> Overview`, the banner should be shown again
12. Install `woocommerce-analytics` plugin, do not activate it
13. Go to `Analytics -> Overview`, the banner should not be shown
14. Activate `woocommerce-analytics` plugin
15. Go to `Analytics -> Overview`, the banner should not be shown 

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ianlin committed:
Implement order attribution install banner in analytics overview page

ianlin committed:
Add a hook for setting banner dismissed option

ianlin committed:
Use the hook the decide whether to show banner and call dismiss funtion
when clicking dismiss CTA

ianlin committed:
Do not show banner when the plugin is already installed

ianlin committed:
Merge branch 'feature/51567-order-attribution-install-and-connect' into add/51567-banner-in-core-analytics-overview

ianlin committed:
Add event tracking for banner viewed and clicked

puntope commented:
**UI Feedback about the text container size:**

I believe that in the paragraph we added a `br` tag just to be in line with the size in the figma file. The problem with this approach is that in responsive looks odd when the "use it to track" gets alone in a single line (see screenshot):

<img width="811" alt="Screenshot 2024-09-25 at 13 44 01" src="https://github.com/user-attachments/assets/de253dc7-2074-48bc-a16d-9e65a55af7a3">

WDYT of removing the `br` and add `max-width` prop in the text container? 

<img width="1550" alt="Screenshot 2024-09-25 at 13 45 22" src="https://github.com/user-attachments/assets/78d59abd-6500-451d-97cc-1d6357b50d00">

puntope commented on plugins/woocommerce-admin/client/order-attribution/order-attribution-install-banner-image.js:
```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
```
Unused dependency

ianlin replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-8dd18925e7fd2b411d66fad20a5b0443e01352e2b8a28ba11fe55775b861071eL4.

puntope commented on plugins/woocommerce-admin/client/order-attribution/index.js:
```diff
@@ -0,0 +1 @@
+export * from './order-attribution-install-banner';
```
I believe this folder pretends to have more order-attributions stuff inside, or just the banner? If it's just the banner I'd suggest rename it to `plugins/woocommerce-admin/client/order-attribution-install-banner/...`

ianlin replied:
Fixed in 30f059e3a9b473309aa9930e5ad308ac8659b46c.

puntope commented on plugins/woocommerce-admin/client/dashboard/index.js:
```diff
@@ -8,6 +8,7 @@ import { Spinner } from '@woocommerce/components';
  * Internal dependencies
  */
 import './style.scss';
+import { OrderAttributionInstallBanner } from '../order-attribution';
```
WDYT of placing the component import before the parent component style?

puntope replied:
💅  `import { OrderAttributionInstallBanner } from '~/order-attribution';` Also works and will work even if the structure changes.

ianlin replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-a158fec6997e1056ba460eecc464f318329f7ee6311a11da74171f4b696bb68cR10.

puntope commented on plugins/woocommerce-admin/client/order-attribution/order-attribution-install-banner.js:
```diff
@@ -0,0 +1,98 @@
+	bannerImage = <OrderAttributionInstallBannerImage />,
+} ) => {
+	const {
+		isOrderAttributionInstallBannerDismissed,
+		dismissOrderAttributionInstallBanner,
```
WDYT about shortening this for better readability?

`isDismissed`, 'dismiss'

ianlin replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-3815cff3f1a08b694ce40b4a0699e0cf64d9f7d0b0f3694313ae78d0bb12f56bR21.

puntope commented on plugins/woocommerce-admin/client/order-attribution/order-attribution-install-banner.js:
```diff
@@ -0,0 +1,98 @@
+		return null;
+	}
+
+	recordEvent( 'order_attribution_install_banner_viewed' );
```
Since bannerImage is a customizable parameter I assume this component will be reusable somewhere else? 

If so, should we add some props for the recordEvents to add information about which of the banners was clicked or viewed?

ianlin replied:
Fixed in

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-3815cff3f1a08b694ce40b4a0699e0cf64d9f7d0b0f3694313ae78d0bb12f56bR28-R31

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-3815cff3f1a08b694ce40b4a0699e0cf64d9f7d0b0f3694313ae78d0bb12f56bL80-R83

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-09863637c86d1cdb04cfa551d822f0ce78dd4589e68783b05e753d90b956da6fL24-R28

puntope commented on plugins/woocommerce-admin/client/order-attribution/order-attribution-install-banner.js:
```diff
@@ -0,0 +1,98 @@
+								'Understand what truly drives revenue with our powerful order attribution extension. Use it to track<br/>your sales journey, identify your most effective marketing channels, and optimize your sales strategy.',
+								'woocommerce'
+							),
+							{ br: <br /> }
```
See my comment [here](https://github.com/woocommerce/woocommerce/pull/51672#issuecomment-2373604960) regarding UI

ianlin replied:
Fixed in

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-3815cff3f1a08b694ce40b4a0699e0cf64d9f7d0b0f3694313ae78d0bb12f56bL67-R70

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-71e16bc648210308ae9a1f2e30c8e8e311fe24425077449417a2f94ca8b2aa9bR60

puntope commented on plugins/woocommerce-admin/client/order-attribution/use-order-attribution-install-banner.js:
```diff
@@ -0,0 +1,68 @@
+} from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+const OPTION_NAME_BANNER_DISMISSED =
```
Should this be just `OPTION_NAME`or `OPTION_BANNER_DISMISSED`?

puntope commented on plugins/woocommerce-admin/client/order-attribution/use-order-attribution-install-banner.js:
```diff
@@ -0,0 +1,68 @@
+	'woocommerce_order_attribution_install_banner_dismissed';
+const OPTION_VALUE_YES = 'yes';
+
+const useOrderAttributionInstallBanner = () => {
```
Can we document this hook more? Like how it works and why it is for.

ianlin replied:
Added in 5c89dee

puntope commented on plugins/woocommerce-admin/client/order-attribution/use-order-attribution-install-banner.js:
```diff
@@ -0,0 +1,68 @@
+
+	return {
+		loading,
+		isOrderAttributionInstallBannerDismissed:
+			isBannerDismissed === OPTION_VALUE_YES,
+		dismissOrderAttributionInstallBanner,
+		shouldShowBanner:
```
Are we using `isOrderAttributionInstallBannerDismissed` for something else than showing the banner or not? 

Just wondering if `isOrderAttributionInstallBannerDismissed` can be out of the exposed props and just part of the `shouldShowBanner` conditional.

ianlin replied:
Yeah, this banner (mZgTgnHhRdyff3Gy5HIwcB-fi-2132_64200) will only be shown when the main banner is dismissed by the merchant.

puntope commented:
Anther UI issue I saw, is that on mobile small screens, the banner is not readable. Even not scrollable.

I belive we don't need a `min-width` here. 

<img width="427" alt="Screenshot 2024-09-25 at 14 42 59" src="https://github.com/user-attachments/assets/ad915fec-8502-4eb5-ad10-3fb3a1768b7e">

puntope commented on plugins/woocommerce-admin/client/order-attribution/style.scss:
```diff
@@ -0,0 +1,61 @@
+	.woocommerce-order-attribution-install-analytics-overview-banner__text-title {
+		color: $studio-gray-90;
+		line-height: 24px;
+		font-weight: 590;
```
I see this is in the Figma as 590 as well. Just curious why design choose a non standard weight instead of 600 for example.

ianlin replied:
Nice catch, changed it to 600 as it's no harm.
https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-71e16bc648210308ae9a1f2e30c8e8e311fe24425077449417a2f94ca8b2aa9bR54

puntope commented on plugins/woocommerce-admin/client/order-attribution/style.scss:
```diff
@@ -0,0 +1,61 @@
+	}
+
+	.woocommerce-order-attribution-install-analytics-overview-banner__text-chip {
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		background: $studio-gray-0;
+		border-radius: 2px;
+		width: 42px;
+		height: 24px;
```
WDYT about relacing 

```
display: inline-flex
justify-content: center;
background: $studio-gray-0;
border-radius: 2px;
padding: 2px 5px; 
```

So the "chip" container can adapt to the text.

puntope replied:
Also, I wonder if "chip" is the best name for this. I believe a wider used name may be "badge" or "label"?

ianlin replied:
Thanks for the tip about styling. The "chip" is from the figma but I also like "badge" more.
Updated in https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-71e16bc648210308ae9a1f2e30c8e8e311fe24425077449417a2f94ca8b2aa9bL37-R44

puntope commented on plugins/woocommerce-admin/client/order-attribution/style.scss:
```diff
@@ -0,0 +1,61 @@
+
+	margin: 0 15px 10px 0;
+	animation: isLoaded;
+	min-width: 550px; // for smaller screens
```
See comment in https://github.com/woocommerce/woocommerce/pull/51672#issuecomment-2373729475

ianlin replied:
Thanks for catching that. I added a media query so the image and the text will be aligned vertically when the screen is smaller.

https://github.com/woocommerce/woocommerce/pull/51672/commits/749fb4dd2a5a4e34933b14b888380db0bc8cf3b3#diff-71e16bc648210308ae9a1f2e30c8e8e311fe24425077449417a2f94ca8b2aa9bR16-R19

puntope commented:
I just noticed also that the event for the banner view is triggering each time the component re-renders


https://github.com/user-attachments/assets/9a0b93ff-09a4-4832-82b9-66fef3bc2657

ianlin committed:
UI adjustments based on PR reviews

ianlin committed:
Move client/order-attribution to client/order-attribution-install-banner

ianlin committed:
Prevent the banner viewed event from being sent again wehn rerendering

ianlin committed:
Merge branch 'feature/51567-order-attribution-install-and-connect' into add/51567-banner-in-core-analytics-overview

ianlin committed:
Add a comment to the hook useOrderAttributionInstallBanner

ianlin commented:
> I just noticed also that the event for the banner view is triggering each time the component re-renders

@puntope Thanks for catching that! I added a useEffect hook to the banner component so it won't be sending the event multiple times when rerendering.

f29120009a50bdc9392131e1ef03904817e724ad

I updated the PR based on all your suggestions, it's ready for another look.

puntope commented on plugins/woocommerce-admin/client/dashboard/index.js:
```diff
@@ -19,6 +20,7 @@ class Dashboard extends Component {
 
 		return (
 			<Suspense fallback={ <Spinner /> }>
+				<OrderAttributionInstallBanner eventContext="analytics-overview" />
```
THis can be omitted as `analytics-overview` is the default value.

puntope commented on plugins/woocommerce-admin/client/order-attribution-install-banner/order-attribution-install-banner.js:
```diff
@@ -0,0 +1,106 @@
+	return (
+		<Card
+			size="medium"
+			className="woocommerce-order-attribution-install-analytics-overview-banner"
```
Since the banner is reusable, WDYT about removing  `-analytics-overview` part?

puntope reviewed (approved):
LGTM ! Thanks for the changes.

Left 2 additional comments but not blockers.

ianlin committed:
Remove the prop of eventContext since it uses the default value

ianlin committed:
Rename the css class name to make it more general
